alternator/rjson.cc

```diff
@@ -128,6 +128,18 @@ std::string print(const rjson::value& value) {
     return std::string(buffer.GetString());
 }
 
+rjson::invalid_parameter::invalid_parameter(std::string_view name, const rjson::value& value)
+    : invalid_parameter(name, print(value))
+{}
+
+rjson::invalid_parameter::invalid_parameter(std::string_view name, std::string_view value)
+    : error(format("Invalid JSON parameter {} : {}", name, value))
+{}
+
+rjson::missing_parameter::missing_parameter(std::string_view name) 
+    : error(format("JSON parameter {} not found", name))
+{}
+
```

Comment:
The exceptions really only serve as message packing but with the extra benefit of being able to distinguish by type in caller catch. I fail to see _why_ you would not want this. The alternative is having copied code that formats the same message in mutiple places, alternatively a subroutine that is called instead of just using a different exception type. C-style. 
As for invalid_parameter, it says that the value could not be parsed as excepted by the (implicit) schema. While it is technically not a syntactical error (valid JSON), it is still a parse error (invalid JSON content).

Reply:
Ok, I hear what you are saying here, and I admit, my using the exceptions above in actual alternator code is plain wrong. But the exception types are actually useful in them selves, mainly to unify message formatting (and keep it in slow path in cc-file) and potentially allow specialized catch (though that is not needed at this point, but...). 

I've changed this in next version so the rjson exceptions are purely rjson. Also added comments about how the existing use of rjson:error already breaks the separation in its message wording.